File: unity-project/Assets/Scripts/CollectableCollision.js
#pragma strict

private var gameController : GameController;
var collectSound : AudioClip;

function Start()
{
	var gameControllerObject : GameObject = GameObject.FindWithTag("GameController");
	if(gameControllerObject != null){
		gameController = gameControllerObject.GetComponent(GameController);
	}
	if(gameControllerObject == null){
		Debug.Log("Can't fint 'GameController' script!");
	}
		
}

function OnCollisionEnter(collision:Collision){
	audio.PlayOneShot(collectSound);
	if(collision.gameObject.tag == "Player" && gameController.isBeamOn()){
		Debug.Log("Hit by player on collision enter");
    	Destroy(this.gameObject);
    	gameController.AddScore(100);
    	gameController.CollectedIncrease(this.gameObject.tag);
    	//Destroy(this);
    	collision.gameObject.SendMessage("HitCollectable");
    }
    
}

File: unity-project/Assets/Scripts/CollectableCollision.cs
using UnityEngine;
using System.Collections;

public class CollectableCollision : MonoBehaviour
{
	private GameController gameController ;
	GameControllerObject gameObject;

	// Use this for initialization
	void Start ()
	{
		gameControllerObject = GameObject.FindWithTag("GameController");
		if(gameControllerObject != null){
			gameController = gameControllerObject.GetComponent(GameController);
		}else{
			Debug.Log("Can't find GameController script!");
		}
	}
	
	// Update is called once per frame
	void Update ()
	{
		if(collision.gameObject.tag == "Player" && gameController.isBeamOn()){
			Debug.Log("Hit by player");
			Destroy(this.gameObject);
			
			gameController.AddScore(100);
			gameController.CollectedIncrease(this.gameObject.tag);
			//Destroy(this);
			collision.gameObject.SendMessage("HitCollectable");
		}
	}
}

